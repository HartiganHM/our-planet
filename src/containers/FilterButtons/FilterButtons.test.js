import React from 'react';
import { shallow } from 'enzyme';
import {
  FilterButtons,
  mapStateToProps,
  mapDispatchToProps,
} from './FilterButtons';

describe('FilterButton tests', () => {
  it('Should match the snapshot', () => {
    const mockFilter = 'default';
    const mockSwitchFilter = jest.fn();
    const renderedFilterButtons = shallow(
      <FilterButtons filter={mockFilter} switchFilter={mockSwitchFilter} />
    );

    expect(renderedFilterButtons).toMatchSnapshot();
  });

  describe('mapStateToProps tests', () => {
    it('Should pull filter from the store', () => {
      const mockStore = {
        filter: 'status',
      };
      const result = mapStateToProps(mockStore);

      expect(result.filter).toEqual(mockStore.filter);
    });
  });

  describe('mapDispatchToProps tests', () => {
    it('Should call dispatch when switchFilter is called', () => {
      const mockDispatch = jest.fn();
      const result = mapDispatchToProps(mockDispatch);

      result.switchFilter('habitat');
      expect(mockDispatch).toHaveBeenCalled();
    });
  });
});
