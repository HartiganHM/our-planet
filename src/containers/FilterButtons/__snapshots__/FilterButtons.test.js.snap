// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterButton tests Should match the snapshot 1`] = `ShallowWrapper {}`;
